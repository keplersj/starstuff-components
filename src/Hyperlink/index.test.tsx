import { fixture } from "atomico/test-dom";
import { Hyperlink } from ".";

customElements.define("starstuff-hyperlink", Hyperlink);

describe("Hyperlink", (): void => {
  it("shadow DOM renders correctly", async (): Promise<void> => {
    const element = fixture(
      <Hyperlink data-testid="starstuff-hyperlink-shadowdom" href="/test">
        Test Link
      </Hyperlink>
    );

    await (element as any).updated;

    const queried = document.querySelector(
      "[data-testid=starstuff-hyperlink-shadowdom]"
    );

    expect(queried).toMatchSnapshot();
    expect(queried?.shadowRoot?.innerHTML).toMatchSnapshot();
  });
});
