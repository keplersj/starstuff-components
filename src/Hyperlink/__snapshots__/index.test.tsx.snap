// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hyperlink shadow DOM renders correctly 1`] = `
<starstuff-hyperlink
  data-testid="starstuff-hyperlink-shadowdom"
  href="/test"
>
  Mark {}
  Test Link
  Mark {}
</starstuff-hyperlink>
`;

exports[`Hyperlink shadow DOM renders correctly 2`] = `
<a href="/test">
  <slot>
  </slot>
</a>
<style>
  :host,
  :host a {
    color: var(--starstuff-hyperlink-default-color);
    text-decoration: none;

    box-shadow: inset 0 -1px 0 0 rgba(var(--starstuff-hyperlink-default-color), 0.5);

    --starstuff-hyperlink-default-color: 0, 0, 0;
  }

  @media screen and (prefers-color-scheme: dark) {
    :host,
    :host a {
      --starstuff-hyperlink-default-color: 255, 255, 255;
    }
  }

  :host(:hover),
  :host(:focus) {
    box-shadow: inset 0 -2px 0 0 rgba(var(--starstuff-hyperlink-default-color), 0.7);
  }
</style>
`;
